src/browser/pages/vscode.ts

```diff
@@ -160,9 +156,72 @@ export function setBodyBackgroundToThemeBackgroundColor(document: Document, loca
   return null
 }
 
-try {
+export function registerLoadBundleOnNlsConfig(nlsConfig: NlsConfiguration, base: string) {
+  const bundles: {
+    [key: string]: string
+  } = {}
+
+  type LoadBundleCallback = (_: undefined, result?: string) => void
+
+  nlsConfig.loadBundle = (bundle: string, _language: string, cb: LoadBundleCallback): void => {
+    const result = bundles[bundle]
+    if (result) {
+      return cb(undefined, result)
+    }
+    // FIXME: Only works if path separators are /.
+    const path = nlsConfig._resolvedLanguagePackCoreLocation + "/" + bundle.replace(/\//g, "!") + ".nls.json"
+    fetch(`${base}/vscode/resource/?path=${encodeURIComponent(path)}`)
+      .then((response) => response.json())
+      .then((json) => {
+        bundles[bundle] = json
+        cb(undefined, json)
+      })
+      .catch(cb)
+  }
+}
+
+/**
+ * A helper function to encapsulate all the
+ * logic used in this file.
+ *
+ * It does the following:
+ * - grabs the options using getOptions
+ * - gets the nlsConfig
+ * - registers loadBundle on nlsConfig
+ * - gets the loader
+ * - sets loader on self.require
+ * - sets the body background color to match the theme
+ *
+ * We purposely include all of this in a single function
+ * so that it's easier to test.
+ */
+export function main() {
+  const options = getOptions()
+  const nlsConfig = getNlsConfiguration(document)
+
+  if (nlsConfig._resolvedLanguagePackCoreLocation) {
+    registerLoadBundleOnNlsConfig(nlsConfig, options.base)
+  }
+
+  const loader = getLoader({
+    nlsConfig,
+    options,
+    origin: window.location.origin,
+  })
+
+  // TODO@jsjoeio
+  // I'm not sure how to properly type cast this
+  // This might be our best bet
+  // Source: https://stackoverflow.com/a/30740935
+  // This object on self.require is what configures the loader
+  // and tells it how to load files that get imported.
+  ;(self.require as FixMeLater) = loader
+
   setBodyBackgroundToThemeBackgroundColor(document, localStorage)
+}
+
+try {
+  main()
```

Comment:
Thoughts on moving globals to the top? So `main(document, window)`, etc? That way we don't have to mess with globals in the unit test.

Reply:
Honestly, that would be my preference! Much easier to unit test. Good call!